### DISCUSSION TITLE
How to use the "post_asset_folder" URL when specifying a cover in front matter?

### DISCUSSION THREAD

**Discussion by Kritner** [ppoffice/hexo-theme-icarus#961] [Q&A]
I'm making use of a `post_asset_folder`, where all my post's images and the like are stored in a folder in source as `YYYYMMDD-my-post-title`.  If I have a cover image, it is in the directory `/source/_posts/20210101-my-post-title/cover.png` as an example. This post then gets generated under `/public/2021/01/01/my-post-title/cover.png` and is hit through the URL `/2021/01/01/my-post-title/cover.png`.

When using a `thumbnail` in the front matter, the `post_asset_folder` is taken into account in such a way that the image source is the full URL to the image, including its subfolders, such as `https://example.com/2021/01/01/my-post-title/cover.png`.  However, when using the `cover` in the front matter, only `https://example.com/cover.png` is being returned as the "image url".  This is causing broken image links on the "cover" but not the "thumbnail", is there a way to get the cover to correctly use the `post_asset_folder` URL construction as well?

**Comment by ppoffice** [comment]
@Kritner  Could you please try the following changes?

At https://github.com/ppoffice/hexo-theme-icarus/blob/4.4.0/layout/common/article.jsx#L36

```diff
- <img class="fill" src={cover} alt={page.title || cover} />
+ <img class="fill" src={url_for(cover)} alt={page.title || cover} />
```


At https://github.com/ppoffice/hexo-theme-icarus/blob/4.4.0/layout/common/article.jsx#L38

```diff
- <img class="fill" src={cover} alt={page.title || cover} />
+ <img class="fill" src={url_for(cover)} alt={page.title || cover} />
```

**Reply by Kritner** [reply]
Thanks @ppoffice I gave it a shot, but it didn't seem to work:

![image](https://user-images.githubusercontent.com/8814983/140666930-830c09a7-d1e6-4df8-9c35-e91f89228829.png)

(Note I did attempt by swapping the line 36/38 like requested, when that didn't work I just logged the output set above to see what I was working with)

I ended up removing the `hexo-featured-image` package, which once doing so has broken the link for `thumbnail` as well:

![image](https://user-images.githubusercontent.com/8814983/140667014-21274391-5a39-46e9-aca4-b2dbcf4efe68.png)

I mention the above before when I was using the `hexo-featured-image` package, `thumbnail` was working, but not `cover` (though featured image expects `featured_image` as the front matter trigger word).  After removing `hexo-featured-image` both `thumbnail` and `cover` are just going to the URL (in the case of running in localhost) `http://localhost:4000/cover.png` rather than my expected `http://localhost:4000/2021/01/01/my-post-title/cover.png`.  

The reproduction of this can be found here: https://github.com/Kritner/HexoFeaturedImageRepro/tree/hexoUpgrade.  Hopefully I've just missed something in my settings, but I have the `post_asset_folder: true`, which I would have expected the `url_for` to account for.  It looks like there was a pretty recent change to [url_for](https://github.com/hexojs/hexo-util/blob/master/lib/url_for.js) in https://github.com/hexojs/hexo-util/pull/262, but I haven't compared the icarus theme's hexo-util version to the version that the mentioned PR would have been released with (assuming it's even been released to NPM at this point).

**Reply by ppoffice** [reply]
Ok. In that case the asset folder feature is simply not supported. Previously, the link worked for `thumbnail` because you had `hexo-featured-image` installed.
